concrete-core/src/specification/engines/glwe_ciphertext_trivial_encryption.rs

```diff
@@ -1,12 +1,31 @@
-use super::engine_error;
-use crate::prelude::PlaintextVectorEntity;
 use concrete_commons::parameters::GlweSize;
 
+use crate::prelude::PlaintextVectorEntity;
 use crate::specification::engines::AbstractEngine;
 use crate::specification::entities::GlweCiphertextEntity;
 
+use super::engine_error;
+
 engine_error! {
     GlweCiphertextTrivialEncryptionError for GlweCiphertextTrivialEncryptionEngine @
+    PlaintextCountMismatch => "The GLWE size must divide the plaintext count of the input \
+                               vector."
+}
+
+impl<EngineError: std::error::Error> GlweCiphertextTrivialEncryptionError<EngineError> {
+    /// Validates the inputs
+    pub fn perform_generic_checks<PlaintextVector>(
+        glwe_size: GlweSize,
+        input: &PlaintextVector,
+    ) -> Result<(), Self>
+    where
+        PlaintextVector: PlaintextVectorEntity,
+    {
+        if (input.plaintext_count().0 % glwe_size.0) != 0 {
+            return Err(Self::PlaintextCountMismatch);
+        }
+        Ok(())
+    }
 }
 
```

Comment:
There is no need for verification here. Since the operator is pure, the output ciphertext will be created with the proper polynomial size. Also, if that was to be checked, we would have to verify the polynomial size, instead of the glwe size (which is the dimension+1).

Reply:
Oh, no sorry: the amount of plaintexts in the GLWE is equal to the polynomial size, right? Ok so I'll remove this check.